src/Common/StackTrace.cpp

```diff
@@ -306,7 +344,11 @@ StackTrace::StackTrace(const ucontext_t & signal_context)
 
 void StackTrace::tryCapture()
 {
+#ifdef __APPLE__
+    size = ::backtrace(frame_pointers.data(), capacity);
```

Comment:
Most likely, it is not async-signal safe, hence cannot be used.

Reply:
Indeed, random page says it is not async-signal safe https://forums.developer.apple.com/forums/thread/113742

But maybe it is still Ok for synchronous unwinding like exceptions?

Reply:
Yes, totally ok for exceptions, not ok for crashes.

Reply:
Profiler is not working on MacOS due to the lack of PHDR cache. So this will be used only for exceptions and places where you write `StackTrace()::toString()`.